bin/src/Main.purs

```diff
@@ -844,23 +845,23 @@ mkPublishEnv dependencies = do
               [ toDoc "No package was selected for publishing. Please select (with -p) one of the following packages:"
               , indent (toDoc $ map _.package.name workspacePackages)
               ]
-  pure (Record.union { selected, dependencies } env)
+  pure (Record.union { selected, packageDependencies } env)
 
-mkReplEnv :: forall a. ReplArgs -> Map PackageName Package -> Spago (Fetch.FetchEnv a) (Repl.ReplEnv ())
-mkReplEnv replArgs dependencies = do
+mkReplEnv :: forall a. ReplArgs -> Map PackageName PackageMap -> PackageMap -> Spago (Fetch.FetchEnv a) (Repl.ReplEnv ())
+mkReplEnv replArgs packageDependencies supportPackage = do
   { workspace, logOptions } <- ask
   logDebug $ "Repl args: " <> show replArgs
 
   purs <- Purs.getPurs
 
-  let
-    selected = case workspace.selected of
-      Just s -> [ s ]
-      Nothing -> Config.getWorkspacePackages workspace.packageSet
+  selected <- case workspace.selected of
+    Just s -> pure $ Build.SinglePackageGlobs s
+    Nothing -> pure $ Build.AllWorkspaceGlobs $ Config.getWorkspacePackages workspace.packageSet
```

Comment:
I think `AllWorkspaceGlobs` always implies that we `getWorkspacePackages`, so we should call this then we go use the value rather than when we construct it (or use a smart constructor)
```suggestion
    Nothing -> pure $ Build.AllWorkspaceGlobs workspace.packageSet
```